pyatrea/__init__.py

```diff
@@ -82,328 +85,390 @@ def decompress(self, s):
+                                    self.params["coefs"][id] = float(
+                                        child.attrib["coef"]
+                                    )
+
+                                if "offset" in child.attrib:
+                                    self.params["offsets"][id] = float(
+                                        child.attrib["offset"]
+                                    )
         return self.params
 
-    def getStatus(self):
-        if not self.status:
-            self.refreshStatus()
-        return self.status
-    
-    def refreshStatus(self):
-        self.status = {}
-        response = requests.get('http://'+self.ip+'/config/xml.xml?auth='+self.code+'&'+random.choice(string.ascii_letters)+random.choice(string.ascii_letters))
+    def getStatus(self, useCache=True):
+        if not self.status or not useCache:
+            self.status = {}
+            response = requests.get(
+                "http://"
+                + self.ip
+                + "/config/xml.xml?auth="
+                + self.code
+                + "&"
+                + random.choice(string.ascii_letters)
+                + random.choice(string.ascii_letters)
+            )
 
-        if(response.status_code == 200):
-            if "HTTP: 403 Forbidden" in response.text:
-                self.auth()
-                response = requests.get('http://'+self.ip+'/config/xml.xml?auth='+self.code+'&'+random.choice(string.ascii_letters)+random.choice(string.ascii_letters))
-                if response.status_code == 200 and "HTTP: 403 Forbidden" in response.text:
-                    return False
-            if(response.status_code == 200):
-                xmldoc = ET.fromstring(response.content)
-                parentData = xmldoc[0] # Known paths to data nodes: /RD5WEB/RD5/ and /PCOWEB/PCO/
-                for data in list(parentData):
-                    for child in list(data):
-                        if(child.tag == "O"):
-                            self.status[child.attrib['I']] = child.attrib['V']
+            if response.status_code == 200:
+                if "HTTP: 403 Forbidden" in response.text:
+                    self.auth()
+                    response = requests.get(
+                        "http://"
+                        + self.ip
+                        + "/config/xml.xml?auth="
+                        + self.code
+                        + "&"
+                        + random.choice(string.ascii_letters)
+                        + random.choice(string.ascii_letters)
+                    )
+                    if (
+                        response.status_code == 200
+                        and "HTTP: 403 Forbidden" in response.text
+                    ):
+                        return False
+                if response.status_code == 200:
+                    xmldoc = ET.fromstring(response.content)
+                    parentData = xmldoc[
```

Comment:
This formatting looks odd. Is it intentional?

Reply:
Yes, line shall not exceed certain number of characters... I will have to check for number and will add lint files to repo later.